### PR TITLE
Updated java examples to use the Gateway

### PR DESCRIPTION
**Pull request by dominikl**
Updates the Java examples to use the Gateway, see https://github.com/openmicroscopy/openmicroscopy/pull/3899

### CODE REVIEW

**Comment by jburel** [comment]
To be considered only when the decoupling work is merged.

**Comment by hflynn** [comment]
Assuming this can be reviewed now unless you have further work to push before it is ready for someone else to look at @dominikl ?

**Comment by dominikl** [comment]
I added some methods for ROI handling to the Gateway, but not tested yet. So I still might have to update the ROI example in the documentation.

**Comment by hflynn** [comment]
Ok, I'll leave you to stick this on stand-up when you are ready then. Thanks!

**Comment by dominikl** [comment]
The documentation should now comply with the Java code examples in Training/Java

**Comment by hflynn** [comment]
Something funny going on with Travis here, any ideas @sbesson ?

**Comment by sbesson** [comment]
Likely related to maintenance activity in the afternoon. Relaunched.

**Comment by hflynn** [comment]
Reference to Requests.java in line 136 of developers/Server/GraphMigration.txt will also need updating here.

**Comment by hflynn** [comment]
Gateway not going in until 5.1.4 now.

**Comment by hflynn** [comment]
Staged at https://www.openmicroscopy.org/site/support/omero5.1-staging/developers/Java.html

**Comment by joshmoore** [comment]
Few comments:

 * I agree in general with what's going on here; comparing to the Python pages, there's no "raw" use of `omero.client()` that we document, so this brings us in line.
 * I do wonder if the code here is or can be tested in any automated way.
 * I missed something from https://github.com/openmicroscopy/openmicroscopy/pull/3952 -- I thought we were going to get rid of the "pojo" package. If so, we should also try to remove use of "pojo" here completely. (That may just be a follow-up PR if we're stating clearly "This is an ALPHA release of this API" etc.)

**Comment by dominikl** [comment]
- The code is basically copied from the Java training examples, but I'm not sure if/where they are run automatically.
- Pojos are going to be removed? Means always working with (slightly annoying, e. g. getXYZ().getValue() etc.) ICE objects directly? I wasn't aware of that...

**Comment by joshmoore** [comment]
> Pojos are going to be removed? Means always working with (slightly annoying, e. g. getXYZ().getValue() etc.) ICE objects directly? I wasn't aware of that...

Sorry, I meant renaming to something more OME-y.

**Comment by jburel** [comment]
code is run as part of training job.

**Comment by hflynn** [comment]
@jburel /gateway/Gateway.java is still showing as a broken link on https://ci.openmicroscopy.org/view/Docs/job/OMERO-DEV-merge-docs/13/warnings3Result/ - is this PR still waiting on further code changes?
Dom is back tomorrow so if you have time to look over this today, he can make any further changes needed and we can get this cleared.

**Comment by jburel** [comment]
@hflynn: I will have a look at it

**Comment by jburel** [comment]
@hflynn: I have opened a PR against @dominikl's PR so he can merged tomorrow when he returns.

File: omero/developers/Java.txt [hunk 1]
```diff
@@ -39,36 +39,41 @@ the dependencies in the ivy.xml files (e.g.
 
 Connect to OMERO
 ----------------
+(**Note:** The usage of the :source:`Gateway.java` <components/blitz/src/omero/gateway/Gateway.java> API is currently still experimental)
```
Comment (hflynn) [inline omero/developers/Java.txt]: Ah! I didn't spot this yesterday but the 2nd ` is in the wrong place, it needs to be after the address in <>:

``:source:`Gateway.java <components/blitz/src/omero/gateway/Gateway.java>``

**Comment by hflynn** [comment]
Sorry @dominikl one slight tweak needed that i missed yesterday

**Comment by hflynn** [comment]
Finally all green thanks! Is this good to merge now @jburel ? Is a card needed on the 5.2 board for any follow on docs re: Josh's comments above?

**Comment by jburel** [comment]
@hflynn: renaming will be another breaking changes. It has been discussed before. note that 3rd party projects already use them.

**Comment by sbesson** [comment]
--no-rebase

**Comment by joshmoore** [comment]
> renaming will be another breaking changes. It has been discussed before. note that 3rd party projects already use them.

Unless it's possible to inject the new renamed classes as superclasses of the existing `pojo.*` classes.

**Comment by joshmoore** [comment]
> Unless it's possible to inject the new renamed classes as superclasses ...

Sorry, this would require multiple inheritance. Sorry for the noise. But I do think this is a question that we should decide very soon since we don't want to be supporting `pojos.*` for any significant amount of time.

**Pull request merged**